docs/eventing/samples/container-source/README.md

```diff
@@ -0,0 +1,133 @@
+ContainerSource will start a container image which will generate events under contain situations and send messages to a sink uri. ContainerSource could be an easy way to implement your own event sources. This document shows how to configure ContainerSource as event source for functions and gives instructions on how to create an event source using ContainerSource.
+
+## Create a heartbeats ContainerSource
+
+### Prerequisites
+
+1. Setup [Knative Serving](../../../serving).
+1. Setup [Knative Eventing and Sources](../../../eventing).
+
+### Prepare the heartbeats image
+
+Knative [event-sources](https://github.com/knative/eventing-sources) has a sample of heartbeats event source. You could clone the source codes by
+```
+git clone https://github.com/knative/eventing-sources.git
+```
+And then build a heartbeats image and publish to your image repo with
+```
+ko publish github.com/knative/eventing-sources/cmd/heartbeats
+```
+**Note**: `ko publish` requires:
+- [`KO_DOCKER_REPO`](https://github.com/knative/serving/blob/master/DEVELOPMENT.md#environment-setup)
+  to be set. (e.g. `gcr.io/[gcloud-project]` or `docker.io/<username>`)
+- you to be authenticated with your `KO_DOCKER_REPO`
+- [`docker`](https://docs.docker.com/install/) to be installed
+
+### Create a Knative Service
+
+In order to verify `ContainerSource` is working, we will create a Event Display Service that dumps incoming messages to its log.
+
+```yaml
+apiVersion: serving.knative.dev/v1alpha1
+kind: Service
+metadata:
+  name: event-display
+spec:
+  runLatest:
+    configuration:
+      revisionTemplate:
+        spec:
+          container:
+            image: gcr.io/knative-releases/github.com/knative/eventing-sources/cmd/event_display
+```
+
+Use following command to create the service from `service.yaml`:
+
+```shell
+kubectl apply --filename service.yaml
+```
+
+### Create a ContainerSource using the heartbeats image
+
+In order to run the heartbeats container as an event source, you have to create a concrete ContainerSource with specific arguments and environment settings. Be sure to replace
+`heartbeats_image_uri` with a valid uri for your heartbeats image in your image repo. Note that arguments and environment variables are set and will be passed to the container.
+
+```yaml
+apiVersion: sources.eventing.knative.dev/v1alpha1
+kind: ContainerSource
+metadata:
+  name: test-heartbeats
+spec:
+  image: <heartbeats_image_uri>
+  sink:
+    apiVersion: serving.knative.dev/v1alpha1
+    kind: Service
+    name: event-display
+  args: 
+    - --period=1
+  env:
+    - name: POD_NAME
+      value: "mypod"
+    - name: POD_NAMESPACE
+      value: "event-test"
+```
+
+Use the following command to create the event source from `heartbeats-source.yaml`:
```

Comment:
That means that we have to write the above YAML sample into a file called `heartbeats-source.yaml`. We should state it clearly.

Reply:
fixed. Thanks.